src/backend/api/controller.js

```diff
@@ -0,0 +1,81 @@
+const knex = require('../database');
+
+const getAllFromTable = async (req, res, table) => {
+  try {
+    const data = await knex(table);
+    res.status(200).json(data);
+  } catch (e) {
+    res.status(500).json(e.message);
+  }
+};
+
+const addToTable = async (req, res, table) => {
+  try {
+    const input = req.body;
+    await knex(table).insert(input);
+    res.status(201).json({ 'message': `${table} added successfuly` });
+  } catch (e) {
+    res.status(500).json(e.message);
+  }
+};
+
+const getById = async (req, res, table) => {
+  try {
+    const data = await knex(table).where(req.params);
+    if (data.length > 0) {
+      res.status(200).json(data);
+    } else {
+      res.status(404).json({
+        'message': `${table} with id ${req.params.id} does not exist`
+      });
+    }
+  } catch (e) {
+    res.status(500).json(e.message);
+  }
+};
+
+const updateById = async (req, res, table) => {
+  try {
+    const id = req.params;
```

Comment:
I really like how you created reusable functions that you use across your code. You did a good job, and since you seem to know what you are doing, I'll push a bit. 

I would suggest changing just passing the req.params to knex(table).where(...), to deconstruct it like this

`const { id } = req.params;
    const data = await knex(table).where({ id });
`
This is recommended for security purposes (SQL injections). 

Other than that, you did a great job.

Reply:
Thanks a lot, will do! 🙌